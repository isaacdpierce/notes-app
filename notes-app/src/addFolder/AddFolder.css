.form__add-folder {
  display: flex;
  justify-content: space-around;
  padding-right: 1rem;
  height: 2.5rem;
}

.form__add-folder input {
  min-width: 6rem;
}

.button__add-folder {
  margin-left: 3px;
  margin-right: 1rem;
}

.button__submit-folder {
  color: var(--main-button-color);
  border: 1px solid var(--main-button-color);
  background-color: transparent;
  font-size: 2rem;
  line-height: 1.2rem;
}

.button__submit-folder:hover {
  color: var(--button-hover-color);
  border: 1px solid var(--button-hover-color);
}
