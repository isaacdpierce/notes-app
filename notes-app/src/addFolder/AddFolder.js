import React, { Component } from 'react';

import './AddFolder.css';

import NotesContext from '../context/NotesContext';

import { config } from '../config/config';

export default class AddFolder extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showForm: false,
      showButton: true,
      folderName: 'Untitled',
    };
  }

  static defaultProps = {
    history: {
      push: () => {},
    },
  };

  static contextType = NotesContext;

  handleAddFormButton() {
    this.setState({
      showForm: true,
      showButton: false,
    });
  }

  handleInput(folderName) {
    this.setState({ folderName });
  }

  handleSubmit = e => {
    e.preventDefault();

    const folder = {
      name: this.state.folderName,
    };

    fetch(`${config.API_ENDPOINT}/folders`, {
      method: 'POST',
      headers: {
        'content-type': 'application/json',
      },
      body: JSON.stringify(folder),
    })
      .then(res => {
        if (!res.ok) return res.json().then(e => Promise.reject(e));
        return res.json();
      })
      .then(folder => {
        this.context.addFolder(folder);
        this.props.history.push(`folder/${folder.id}`);
      })
      .catch(error => {
        console.error({ error });
      });

    this.setState({
      showButton: true,
      showForm: false,
      folderName: 'Untitled',
    });
  };

  render() {
    const { showForm, showButton } = this.state;
    return (
      <>
        {showButton && (
          <>
            <button
              className='note__button button__add-folder '
              onClick={() => this.handleAddFormButton()}
            >
              Add Folder
            </button>
          </>
        )}
        {showForm && (
          <form className='form__add-folder'>
            <input
              type='text'
              id='folder-name-input'
              name='folder-name'
              defaultValue='Untitled'
              onChange={e => this.handleInput(e.target.value)}
            />
            <button
              className='button__submit-folder'
              type='submit'
              onClick={e => this.handleSubmit(e)}
            >
              +
            </button>
          </form>
        )}
      </>
    );
  }
}
