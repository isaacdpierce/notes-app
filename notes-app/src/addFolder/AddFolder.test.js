import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import AddFolder from './AddFolder';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

it('renders the complete form', () => {
  const wrapper = shallow(<AddFolder />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
