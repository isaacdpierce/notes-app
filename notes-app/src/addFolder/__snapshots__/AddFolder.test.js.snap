// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the complete form 1`] = `
<Fragment>
  <button
    className="note__button button__add-folder "
    onClick={[Function]}
  >
    Add Folder
  </button>
</Fragment>
`;
